core/domain/app_feedback_report_validators.py

```diff
@@ -0,0 +1,400 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for app feedback report models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+
+from core.domain import base_model_validators
+from core.platform import models
+import feconf
+
+(
+    base_models, app_feedback_report_models
+) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.app_feedback_report
+])
+
+# Timestamp in sec since epoch for Mar 1 2021 12:00:00 UTC.
+EARLIEST_VALID_DATETIME = datetime.datetime.fromtimestamp(1614556800)
+
+
+class AppFeedbackReportModelValidator(base_model_validators.BaseModelValidator):
+    """Class for validating AppFeedbackReportModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        # Valid id: [platform].[submission_timestamp_in_sec_int].[random_hash]
+        regex_string = '^%s\\.%s\\.[A-Za-z0-9]{1,%s}$' % (
+            item.platform, item.submitted_on.second, base_models.ID_LENGTH)
```

Comment:
Oh you're right, I definitely misunderstood that in the docs, I should be using timestamp() here.

Reply:
So timestamp() is only in Python 3 -- I'm going to pivot to use isoformat() instead for these cases since it seems like we're still in-between 2 & 3. WDYT?

Reply:
After a follow-up, will use milliseconds to match current convention for other entity_ids.